src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Windows.targets

```diff
@@ -24,4 +24,16 @@ Copyright (c) .NET Foundation. All rights reserved.
   <PropertyGroup Condition="'$(TargetPlatformIdentifier)' == 'windows' and '$(TargetPlatformVersion)' == ''" >
     <TargetPlatformVersion>7.0</TargetPlatformVersion>
   </PropertyGroup>
+
+  <Target Name="_CheckForInvalidWindowsTargetPlatformVersion"
+        BeforeTargets="_CheckForInvalidConfigurationAndPlatform"
+        Condition="'$(TargetPlatformIdentifier)' == 'Windows' and '$(TargetPlatformVersion)' != '' and '@(SupportedTargetPlatform)' != ''">
+    <ItemGroup>
+      <ValidWindowsTargetPlatformVersion Include="@(SupportedTargetPlatform)" Condition="'%(Identity)' == '$(TargetPlatformVersion)'" />
+    </ItemGroup>
+    
+    <NetSdkError Condition="'@(ValidWindowsTargetPlatformVersion)' == ''"
+                 ResourceName="InvalidWindowsTargetPlatformVersion"
+                 FormatArguments="$(TargetPlatformVersion)" />
+  </Target>
```

Comment:
I think we should make this logic generic for all target platforms.  Follow the same pattern as we do for checking the `TargetPlatformIdentifier` in #12202.  Move the target out of this file, and generate the error if a `TargetPlatformVersionSupported` property isn't set to true.  Set that property to true (also in the generic targets) if it isn't already set and TargetPlatformVersion matches any of the `SupportedTargetPlatform` items.